src/utils/profile.ts

```diff
@@ -0,0 +1,71 @@
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import { ResponseErrorInternal } from "@pagopa/ts-commons/lib/responses";
+import { pipe } from "fp-ts/lib/function";
+import { User } from "../../src/types/user";
+import ProfileService from "../../src/services/profileService";
+import { InitializedProfile } from "../../generated/backend/InitializedProfile";
+import { EmailAddress } from "../../generated/io-api/EmailAddress";
+
+// define a type that represents a Profile with a non optional email address
+const ProfileWithEmailAddress = t.intersection([
+  t.interface({
+    email: EmailAddress
+  }),
+  InitializedProfile
+]);
+
+type ProfileWithEmailAddress = t.TypeOf<typeof ProfileWithEmailAddress>;
+
+// define a branded type that ensure we have the email in the profile and
+// that it is not undefined
+interface IProfileWithValidNameAndEmailAddressTag {
+  readonly HasValidEmailAddress: unique symbol;
+}
+
+const ProfileWithValidNameAndEmailAddress = t.brand(
+  ProfileWithEmailAddress,
+  (
+    p
+  ): p is t.Branded<
+    ProfileWithEmailAddress,
+    IProfileWithValidNameAndEmailAddressTag
+  > => !!(p.email && p.is_email_validated),
+  "HasValidEmailAddress"
+);
+
+type ProfileWithValidNameAndEmailAddress = t.TypeOf<
+  typeof ProfileWithValidNameAndEmailAddress
+>;
+
+/**
+ * Gets a profile with a valid email
+ *
+ * @param profileService
+ * @param user
+ * @returns
+ */
+export const profileWithValidNameAndEmailAddressOrError = (
```

Comment:
ditto. The name of this should become something like
```suggestion
export const profileWithEmailValidatedOrError = (
```

Reply:
fixed in https://github.com/pagopa/io-backend/pull/926/commits/ca533a71f9f89340555f1e21f73156c352de4b6c